eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/MigrationMethods.scala

```diff
@@ -0,0 +1,16 @@
+package fr.acinq.eclair.wire.internal.channel
+
+import fr.acinq.bitcoin.scalacompat.Transaction
+import fr.acinq.eclair.channel.{Commitments, DATA_WAIT_FOR_FUNDING_CONFIRMED}
+import fr.acinq.eclair.wire.protocol.{ChannelReady, FundingCreated, FundingSigned}
+import fr.acinq.eclair.{Alias, BlockHeight}
+import shapeless.{::, HNil}
+
+object MigrationMethods {
+
+  def genLocalAlias: Commitments :: Option[Transaction] :: BlockHeight :: Option[ChannelReady] :: Either[FundingCreated, FundingSigned] :: HNil => DATA_WAIT_FOR_FUNDING_CONFIRMED = {
+    case commitments :: fundingTx :: waitingSince :: deferred :: lastSent :: HNil =>
+      DATA_WAIT_FOR_FUNDING_CONFIRMED(commitments, fundingTx, waitingSince, Alias(commitments.channelId.take(32).toLong()), deferred, lastSent)
+  }
```

Comment:
Tried this as an experiment. It allows us to factor codec migrations across all versions, and is safe since the compiler enforces a strict type check.

It should maybe return a `HList` though, so we can stack migrations.

Reply:
Done in [5f3fa19](https://github.com/ACINQ/eclair/pull/2316/commits/5f3fa197317b862812a34141311f3bdee9e757b5).